generator/internal/metadata_decorator_generator.cc

```diff
@@ -406,6 +406,11 @@ void $metadata_class_name$::SetMetadata(grpc::ClientContext& context,
   }
   auto const& authority = options.get<AuthorityOption>();
   if (!authority.empty()) context.set_authority(authority);
+  if (options.has<google::cloud::CustomHeadersOption>()) {
```

Comment:
Optional nit: You could do without the `has<T>()` test (and hence the double lookup) in this case, at the cost of a once-only allocation of an empty map.